### ISSUE TITLE
Unable to install on Mingw64 environement

### ISSUE THREAD

**Issue by shadow2560** [BrianPugh/gnwmanager#154]
Trying install on Mingw64 environement but the "pipx install 
gnwmanager" command doesn't work, here is the log of the installation:
```
PIP STDOUT
----------
Collecting gnwmanager
  Using cached gnwmanager-0.10.4-py3-none-any.whl.metadata (8.9 kB)
Collecting autoregistry<2.0.0,>=1.0.0 (from gnwmanager)
  Using cached autoregistry-1.1.2-py3-none-any.whl.metadata (5.4 kB)
Collecting cyclopts<3.0.0,>=2.0.0 (from gnwmanager)
  Using cached cyclopts-2.6.1-py3-none-any.whl.metadata (11 kB)
Collecting littlefs-python>=0.9.0 (from gnwmanager)
  Using cached littlefs_python-0.10.3.tar.gz (486 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting pillow<11.0.0,>=10.0.0 (from gnwmanager)
  Using cached pillow-10.3.0.tar.gz (46.6 MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Installing backend dependencies: started
  Installing backend dependencies: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting psutil<6.0.0,>=5.9.6 (from gnwmanager)
  Using cached psutil-5.9.8.tar.gz (503 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting pyelftools>=0.29 (from gnwmanager)
  Using cached pyelftools-0.31-py3-none-any.whl.metadata (381 bytes)
Collecting pyocd>=0.35.1 (from gnwmanager)
  Using cached pyocd-0.36.0-py3-none-any.whl.metadata (11 kB)
Collecting rich<14.0.0,>=13.5.2 (from gnwmanager)
  Using cached rich-13.7.1-py3-none-any.whl.metadata (18 kB)
Collecting tamp<2.0.0,>=1.1.6 (from gnwmanager)
  Using cached tamp-1.4.1.tar.gz (37 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting tqdm<5.0.0,>=4.66.1 (from gnwmanager)
  Using cached tqdm-4.66.2-py3-none-any.whl.metadata (57 kB)
Collecting typing-extensions<5.0.0,>=4.7.1 (from gnwmanager)
  Using cached typing_extensions-4.11.0-py3-none-any.whl.metadata (3.0 kB)
Collecting attrs>=23.1.0 (from cyclopts<3.0.0,>=2.0.0->gnwmanager)
  Using cached attrs-23.2.0-py3-none-any.whl.metadata (9.5 kB)
Collecting docstring-parser<0.16,>=0.15 (from cyclopts<3.0.0,>=2.0.0->gnwmanager)
  Using cached docstring_parser-0.15-py3-none-any.whl.metadata (2.4 kB)
Collecting rich-rst<2.0.0,>=1.2.0 (from cyclopts<3.0.0,>=2.0.0->gnwmanager)
  Using cached rich_rst-1.3.0-py3-none-any.whl.metadata (6.0 kB)
Collecting capstone<5.0,>=4.0 (from pyocd>=0.35.1->gnwmanager)
  Using cached capstone-4.0.2.tar.gz (2.0 MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Installing backend dependencies: started
  Installing backend dependencies: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting cmsis-pack-manager<1.0,>=0.5.2 (from pyocd>=0.35.1->gnwmanager)
  Using cached cmsis_pack_manager-0.5.3.tar.gz (55 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'error'

PIP STDERR
----------
  error: subprocess-exited-with-error
  
  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: 1
  ╰─> [56 lines of output]
      Collecting maturin<0.15,>=0.13
        Using cached maturin-0.14.17.tar.gz (177 kB)
        Installing build dependencies: started
        Installing build dependencies: finished with status 'done'
        Getting requirements to build wheel: started
        Getting requirements to build wheel: finished with status 'done'
        Preparing metadata (pyproject.toml): started
        Preparing metadata (pyproject.toml): finished with status 'done'
      Collecting cffi
        Using cached cffi-1.16.0.tar.gz (512 kB)
        Installing build dependencies: started
        Installing build dependencies: finished with status 'done'
        Getting requirements to build wheel: started
        Getting requirements to build wheel: finished with status 'error'
        error: subprocess-exited-with-error
      
        Ã— Getting requirements to build wheel did not run successfully.
        â”‚ exit code: 1
        â•°â”€> [27 lines of output]
            Traceback (most recent call last):
              File "C:/Users/user_name/pipx/shared/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
                main()
              File "C:/Users/user_name/pipx/shared/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
                json_out['return_val'] = hook(**hook_input['kwargs'])
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
              File "C:/Users/user_name/pipx/shared/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
                return hook(config_settings)
                       ^^^^^^^^^^^^^^^^^^^^^
              File "C:/msys2/tmp/pip-build-env-o4fvl4b9/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
                return self._get_build_requires(config_settings, requirements=['wheel'])
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
              File "C:/msys2/tmp/pip-build-env-o4fvl4b9/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
                self.run_setup()
              File "C:/msys2/tmp/pip-build-env-o4fvl4b9/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 311, in run_setup
                exec(code, locals())
              File "<string>", line 126, in <module>
              File "<string>", line 105, in uses_msvc
              File "C:/msys2/tmp/pip-build-env-o4fvl4b9/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/config.py", line 220, in try_compile
                self._compile(body, headers, include_dirs, lang)
              File "C:/msys2/tmp/pip-build-env-o4fvl4b9/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/config.py", line 132, in _compile
                self.compiler.compile([src], include_dirs=include_dirs)
              File "C:/msys2/tmp/pip-build-env-o4fvl4b9/overlay/lib/python3.11/site-packages/setuptools/_distutils/_msvccompiler.py", line 343, in compile
                self.initialize()
              File "C:/msys2/tmp/pip-build-env-o4fvl4b9/overlay/lib/python3.11/site-packages/setuptools/_distutils/_msvccompiler.py", line 246, in initialize
                raise DistutilsPlatformError(
            distutils.errors.DistutilsPlatformError: --plat-name must be one of ('win32', 'win-amd64', 'win-arm32', 'win-arm64')
            [end of output]
      
        note: This error originates from a subprocess, and is likely not a problem with pip.
      error: subprocess-exited-with-error
      
      Ã— Getting requirements to build wheel did not run successfully.
      â”‚ exit code: 1
      â•°â”€> See above for output.
      
      note: This error originates from a subprocess, and is likely not a problem with pip.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

```

If I install it on a Windows Python environement it work fine.

**Comment by BrianPugh** [comment]
This should be fixed in the newly released v0.11.0.

Basically, I just removed `pyocd` as a mandatory dependency, as most people use openocd anyways.

**Comment by shadow2560** [comment]
With this new version it install succesfuly but need some settings to the environement.

```
pacman -S mingw-w64-x86_64-gcc mingw-w64-x86_64-make mingw-w64-x86_64-python mingw-w64-x86_64-python-pip mingw-w64-x86_64-python-lz4 mingw-w64-x86_64-python-pillow mingw-w64-x86_64-python-tqdm mingw-w64-x86_64-python-yaml mingw-w64-x86_64-python-pyzopfli mingw-w64-x86_64-python-pyelftools mingw-w64-x86_64-python-pycryptodome mingw-w64-x86_64-python-keystone mingw-w64-x86_64-python-colorama mingw-w64-x86_64-python-numpy mingw-w64-x86_64-python-psutil mingw-w64-x86_64-openocd
SETUPTOOLS_USE_DISTUTILS=stdlib pip install gnwmanager
```

I may have indicated some unusful packages but they are usful for some projects with the GnW.

I've also retried with 0.10.0 version but even with the rust compiler and capstone and cmake packages installed via Pacman it doesn't want to compile Capstone (I don't know why it try because a version is installed via pacman, maybe Pip version is prefered here) and hidapi. I put the end of the log with the errors but not realy important for me now because I use Openocd and the environement is far easier to config with the new 0.11.0 build:
```
  × Building wheel for capstone (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [27 lines of output]
      running bdist_wheel
      running build
      Building C extensions
      CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
        Compatibility with CMake < 3.5 will be removed from a future version of
        CMake.

        Update the VERSION argument <min> value or use a ...<max> suffix to tell
        CMake that the project does not need compatibility with older versions.


      CMake Error at CMakeLists.txt:2 (project):
        Running

         'nmake' '-?'

        failed with:

         no such file or directory


      CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
      CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
      -- Configuring incomplete, errors occurred!
      'nmake' n'est pas reconnu en tant que commande interne
      ou externe, un programme ex‚cutable ou un fichier de commandes.
      error: [Errno 2] No such file or directory: 'capstone.dll'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for capstone
  Building wheel for cmsis-pack-manager (pyproject.toml) ... done
  Created wheel for cmsis-pack-manager: filename=cmsis_pack_manager-0.5.3-py3-none-win_amd64.whl size=4670598 sha256=5547112d203c9f25624f549eedb363300088dd7edfbfb69c5ec3723f8d8348a7
  Stored in directory: c:/users/user_name/appdata/local/pip/cache/wheels/f4/6d/1e/9719ff47dfd9d4976181b730b91ef739e6fd4c547c18e64d07
  Building wheel for hidapi (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for hidapi (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [9 lines of output]
      running bdist_wheel
      running build
      running build_ext
      C:/msys2/tmp/pip-build-env-uopudz8w/normal/lib/python3.11/site-packages/Cython/Compiler/Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: C:/dev/game_and_watch/Ultimate-GNW-Hack-Script/msys2/tmp/pip-install-pehmsnqk/hidapi_56cde747fc1a49eca2b1fea1edb036d3/hid.pyx
        tree = Parsing.p_module(s, pxd, full_module_name)
      Compiling hid.pyx because it changed.
      [1/1] Cythonizing hid.pyx
      building 'hid' extension
      error: unknown file type '.pxd' (from 'chid.pxd')
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for hidapi
Successfully built cmsis-pack-manager
Failed to build capstone hidapi
ERROR: Could not build wheels for capstone, hidapi, which is required to install pyproject.toml-based projects
```

Thanks.
